### PR TITLE
auto-publish major, minor docker tags

### PR DESCRIPTION
**Pull request by jonathanmeier5**
Resolves openslate/openslate#1894

Feature:
Adds a `--tag-major-minor` flag to the `publish` that will automatically tag major and minor versions of an image.
e.g for image `my-repository.foobar.org/my-cool-image:1.2.3`:
```
cf -e prod publish --tag-major-minor
```
publishes:
```
`my-repository.foobar.org/my-cool-image:1.2.3`
`my-repository.foobar.org/my-cool-image:1.2`
`my-repository.foobar.org/my-cool-image:1`
```

### CODE REVIEW

File: src/compose_flow/image.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+
+OFFICIAL_RELEASE_REGEX = re.compile(r'^\d+\.\d+\.\d+$')
+
+
+class AbstractImage:
+    pass
+
+
+class PrivateImage(AbstractImage):
```
Comment (rca) [inline src/compose_flow/image.py]: Is there a reason for a distinction between a private image and a public one for publishing?

Comment (jonathanmeier5) [reply src/compose_flow/image.py]: If we do decide to add business rules, I think they're meaningfully different for public vs private images.

Comment (rca) [reply src/compose_flow/image.py]: You’re saying the logic for publishing `major` and `major.minor` tags would change based on whether the image is public or private?

Comment (jonathanmeier5) [reply src/compose_flow/image.py]: Out of an offline conversation, we decided that placing support for potential future feature logic obfuscates the existing code's purpose. If we do ever have a use case where we need to apply business rules to public/private images, we can make this distinction then.

**Comment by jonathanmeier5** [comment]
> Overall this is great
> 
> I'm not super jazzed about the argument name `--auto-tag` but I can't think of a clearer one 😬

Yeah, agreed. I changed the flag to `--tag-major-minor`. Not great either, but clearer than before.

**Pull request merged**